apps/console/src/features/applications/pages/application-edit.tsx

```diff
@@ -224,6 +224,18 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
             return;
         }
 
+        /**
+         * If there's no application templateId in the application model,
+         * then we manually bind a templateId. You may ask why templateId is
+         * null at this point? Well, one reason is that, if you create an
+         * application via the API, the templateId is an optional property
+         * in the model. So, if someone creates one without it, we don't have
+         * a template to bootstrap the model.
+         */
+        if (!application?.templateId) {
+            application.templateId = ApplicationManagementConstants.CUSTOM_APPLICATION_OIDC;
```

Comment:
Is this just for OIDC apps or will the same apply to SAML and other type of apps as well?

Reply:
This is addressed with [c4d2b9b](https://github.com/wso2/identity-apps/pull/3040/commits/c4d2b9bdacd75c64b0be797e8aa939974f2285ff). Now it looks for the type of inbound protocol and switches the template based on that.